### PR TITLE
[13.0][ADD] sale_invoicing_date_selection: New module to set invoice_date when user creates invoices from wizard

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa

ping @carlosdauden @CarlosRoca13

### CODE REVIEW

**Review by carlosdauden — APPROVED** [review]

**Review by CarlosRoca13 — APPROVED** [review]
Functional and code review

File: sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            self.env["sale.advance.payment.inv"].with_context(
+                active_model="sale.order", active_ids=orders.ids, open_invoices=True
+            )
+        ) as wiz_invoice_form:
+            wiz_invoice_form.invoice_date = fields.Date.today()
+        wiz = wiz_invoice_form.save()
+        action = wiz.create_invoices()
+        invoices = self.env["account.move"].search(action["domain"])
+        self.assertTrue(invoices[0].invoice_date)
+        self.assertTrue(invoices[1].invoice_date)
```
Comment (pedrobaeza) [inline sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py]: Put here a fixed date and check such date on the test:

```suggestion
            wiz_invoice_form.invoice_date = "2022-07-01"
        wiz = wiz_invoice_form.save()
        action = wiz.create_invoices()
        invoices = self.env["account.move"].search(action["domain"])
        self.assertEqual(invoices[0].invoice_date, fields.Date.from_string("2022-07-01"))
        self.assertEqual(invoices[1].invoice_date, fields.Date.from_string("2022-07-01"))
```

File: sale_invoicing_date_selection/wizard/sale_advance_payment_inv.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    invoice_date = fields.Date(string="Invoice Date")
+
+    def create_invoices(self):
+        """
+        Add date to crate all invoices into user context
+        """
+        return super(
+            SaleAdvancePaymentInv,
+            self.with_context(default_invoice_date=self.invoice_date),
```
Comment (pedrobaeza) [inline sale_invoicing_date_selection/wizard/sale_advance_payment_inv.py]: Isn't there a problem with the accounting date if we force this?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py (hunk 1 above)
Comment (sergio-teruel) [reply sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py]: This changes are not needed!!

Comment (pedrobaeza) [reply sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py]: You are not really testing the real thing. Asserting if there is a date is not the same as asserting that the date you have put is the one that is in the invoice.

File: sale_invoicing_date_selection/wizard/sale_advance_payment_inv.py (hunk 2 above)
Comment (sergio-teruel) [reply sale_invoicing_date_selection/wizard/sale_advance_payment_inv.py]: Chage done!! Now the same behavior is maintained in the same way that I set the manual date in the invoice

File: sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py (hunk 1 above)
Comment (sergio-teruel) [reply sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py]: Yes.. I know!!
But only need that is not empty

Comment (pedrobaeza) [reply sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py]: But in next versions or with any patch a default date is filled and this mechanism won't work, and tests won't detect the problem being so relaxed in this condition. Why not doing what I'm talking about?

Comment (sergio-teruel) [reply sale_invoicing_date_selection/tests/test_sale_invoicing_date_selection.py]: Changes done

**Comment by pedrobaeza** [comment]
You need to use `fields.Date.from_string`, or tests won't be correct.

**Comment by sergio-teruel** [comment]
@pedrobaeza Done!

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**